packages/flutter/test/cupertino/search_field_test.dart

```diff
@@ -0,0 +1,475 @@
+    (WidgetTester tester) async {
+      String value = 'text entry';
+      final TextEditingController controller = TextEditingController();
+      await tester.pumpWidget(
+        CupertinoApp(
+          home: Center(
+            child: CupertinoSearchTextField(
+              controller: controller,
+              placeholder: 'placeholder',
+              onChanged: (String newValue) => value = newValue,
+            ),
+          ),
+        ),
+      );
+
+      controller.text = value;
+      await tester.pump();
+
+      await tester.tap(find.byIcon(CupertinoIcons.xmark_circle_fill));
+      await tester.pump();
+
+      expect(controller.text, isEmpty);
+      expect(find.text('text entry'), findsNothing);
+      expect(value, isEmpty);
+    },
+  );
+
+  testWidgets(
+    'RTL puts attachments to the right places',
+    (WidgetTester tester) async {
+      await tester.pumpWidget(
+        const CupertinoApp(
+          home: Directionality(
+            textDirection: TextDirection.rtl,
+            child: Center(
+              child: CupertinoSearchTextField(
+                suffixMode: OverlayVisibilityMode.always,
+              ),
+            ),
+          ),
+        ),
+      );
+
+      expect(
+        tester.getTopLeft(find.byIcon(CupertinoIcons.search)).dx,
+        800.0 - 26.0,
+      );
+
+      expect(
+        tester.getTopRight(find.byIcon(CupertinoIcons.xmark_circle_fill)).dx,
+        25.0,
+      );
+    },
+  );
+
+  testWidgets(
+    'Can modify prefix and suffix insets',
+    (WidgetTester tester) async {
+      await tester.pumpWidget(
+        const CupertinoApp(
+          home: Center(
+            child: CupertinoSearchTextField(
+              suffixMode: OverlayVisibilityMode.always,
+              prefixInsets: EdgeInsets.all(0),
+              suffixInsets: EdgeInsets.all(0),
+            ),
+          ),
+        ),
+      );
+
+      expect(
+        tester.getTopLeft(find.byIcon(CupertinoIcons.search)).dx,
+        0.0,
+      );
+
+      expect(
+        tester.getTopRight(find.byIcon(CupertinoIcons.xmark_circle_fill)).dx,
+        800.0,
+      );
+    },
+  );
+
+  testWidgets(
+    'custom suffix onTap overrides default clearing behavior',
+    (WidgetTester tester) async {
+      String value = 'text entry';
+      final TextEditingController controller = TextEditingController();
+      await tester.pumpWidget(
+        CupertinoApp(
+          home: Center(
+            child: CupertinoSearchTextField(
+              controller: controller,
+              onChanged: (String newValue) => value = newValue,
+              onSuffixTap: () {},
+            ),
+          ),
+        ),
+      );
+
+      controller.text = value;
+      await tester.pump();
+
+      await tester.tap(find.byIcon(CupertinoIcons.xmark_circle_fill));
+      await tester.pump();
+
+      expect(controller.text, isNotEmpty);
+      expect(find.text('text entry'), findsOneWidget);
+      expect(value, isNotEmpty);
```

Comment:
it's a bit circular, I'm not sure if we're actually exercising the onChanged here